AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -3819,7 +3822,79 @@ function Debug-AzStorageAccountEntraKerbAuth {
                 Write-Error $_
             }
         }
-
+        #
+        # AAD Object check 
+        #
+        if (!$filterIsPresent -or $Filter -match "CheckAADObject")
+        {
+            try {
+                $checksExecuted += 1;
+                Write-Verbose "CheckADObject - START"
+                $Context = Get-AzContext
+                $TenantId = $Context.Tenant
+                Connect-MgGraph -Scopes "Application.Read.All" -TenantId $TenantId
+                Get-MgApplication -Filter "identifierUris/any (uri:uri eq 'api://${TenantId}/CIFS/${StorageAccountName}.file.core.windows.net')" -ConsistencyLevel eventual
+                $SPNValue = Get-MgServicePrincipal -Filter "servicePrincipalNames/any (name:name eq 'api://$TenantId/CIFS/$StorageAccountName.file.core.windows.net')" -ConsistencyLevel eventual

```

Comment:
We may need to add a check on what happens if it's not found, e.g. if the returned value is $null.

Reply:
Also, I think it would make sense to call this variable $ServicePrincipal

Reply:
Fixed